dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.cpp

```diff
@@ -436,28 +436,31 @@ ModelBus::iiAdd(const double& ii) {
 }
 
 void
-ModelBus::evalYType() {
-  yType_[0] = ALGEBRAIC;
-  yType_[1] = ALGEBRAIC;
+ModelBus::evalStaticYType() {
+  if (hasConnection_) {
+    yType_[2] = ALGEBRAIC;
+    yType_[3] = ALGEBRAIC;
+  }
+}
 
+void
+ModelBus::evalDynamicYType() {
   if (hasDifferentialVoltages_ && !getSwitchOff()) {
     yType_[0] = DIFFERENTIAL;
     yType_[1] = DIFFERENTIAL;
-  }
-
-  if (!network_->isInitModel() && hasConnection_) {
-    yType_[2] = ALGEBRAIC;
-    yType_[3] = ALGEBRAIC;
+  } else {
+    yType_[0] = ALGEBRAIC;
+    yType_[1] = ALGEBRAIC;
   }
 }
```

Comment:
This is done in the initialization of the KinAlgRestoration solver, which is called in CalculateIC, do you think it is enough?

Reply:
I figured out this very lately at the end of my review but didn't modify my previous comments. They show that even with a good knowledge of the overall Dynawo process, it is not easy to find the place where the evalDynamicYType is done during the initialization process.
In addition, if the initialization process evolves, it is very plausible that this will be forgotten as it is a kind of hidden feature of the call to SolverKINAlgBlabla. 

Considering this, I would add an explicit call to evalDynamicYType in initBuffers even if it is then evaluate again during the call to the solver.

Reply:
This is not possible as evalDynamicYType uses Y values to be evaluated so we need at least the local init to be done before calling it

Reply:
Yes, your are of course right as we need to be able to test the condition.
Then, I don't know exactly where (in an existing method or in creating a new method) but I would like to see it more explicitly in the initialization scheme instead of hidden in the solver call.

Reply:
done